website/docs/d/custom_attribute.html.markdown

```diff
@@ -0,0 +1,38 @@
+---
+layout: "vsphere"
+page_title: "VMware vSphere: vsphere_custom_attribute"
+sidebar_current: "docs-vsphere-data-source-custom-attriubte"
+description: |-
+  Provides a vSphere custom attribute data source. This can be used to reference custom attributes not managed in Terraform.
+---
+
+# vsphere\_custom\_attribute
+
+The `vsphere_custom_attribute` data source can be used to reference custom 
+attributes that are not managed by Terraform. Its attributes are exactly the 
+same as the [`vsphere_custom_attribute` resource][resource-custom-attribute], 
+and, like importing, the data source takes a name to search on. The `id` and 
+other attributes are then populated with the data found by the search.
+
+[resource-custom-attribute]: /docs/providers/vsphere/r/custom_attribute.html
+
+~> **NOTE:** Custom attributes are unsupported on direct ESXi connections 
+and require vCenter.
+
+## Example Usage
+
+```hcl
+data "vsphere_custom_attribute" "attribute" {
+  name = "terraform-test-attribute"
+}
+```
+
+## Argument Reference
+
+* `name` - (String, required) The name of the custom attribute.
```

Comment:
We recently standardized on a new naming convention for attributes that I want to make sure is followed - the [VM resoruce](https://www.terraform.io/docs/providers/vsphere/r/virtual_machine.html#argument-reference) has a good layout of the format. An example is below:

> `some_attr` - (Required or Optional) Does some thing. Forces a new resource if changed (if this actually is a `ForceNew`) argument.